src/snowflake/snowpark/column.py

```diff
@@ -226,9 +227,16 @@ class Column:
     """
 
     def __init__(
-        self, expr1: Union[str, Expression], expr2: Optional[str] = None
+        self,
+        expr1: Union[str, Expression],
+        expr2: Optional[str] = None,
+        ast: Optional[proto.SpColumnExpr] = None,
     ) -> None:
+        if ast is not None:
```

Comment:
This should be done without a guard. It's better to rely on a known slot being set to `None` rather than the absence of the key in the dictionary.